streams/src/test/java/org/apache/kafka/streams/kstream/internals/CogroupedKStreamImplTest.java

```diff
@@ -173,6 +174,56 @@ public void shouldNameProcessorsAndStoreBasedOnNamedParameter() {
                 "      <-- KTABLE-TOSTREAM-0000000005\n\n"));
     }
 
+    @Test
+    public void shouldNameProcessorsAndStoreBasedOnNamedParameterRepartition() {
+        final KStream<String, String> stream1 = builder.stream("one", stringConsumed);
+        final KStream<String, String> test2 = builder.stream("two", stringConsumed);
+
+        final KGroupedStream<String, String> groupedOne = stream1.map((k, v) -> new KeyValue<>(v, k)).groupByKey();
+        final KGroupedStream<String, String> groupedTwo = test2.groupByKey();
+
+        final KTable<String, String> customers = groupedOne
+                .cogroup(STRING_AGGREGATOR)
+                .cogroup(groupedTwo, STRING_AGGREGATOR)
+                .aggregate(STRING_INITIALIZER, Named.as("test"), Materialized.as("store"));
```

Comment:
nit: no need to specify names, as this test verifies that we insert repartition topics correctly. (I basic test principle is to only do the minimum to express the test -- otherwise, it may be distracting from the purpose of the test, ie, the reader ask, why did we specify those name? How is it related?)